docs: bash in user job images (#146)
reanahub/docs.reana.io: docs/advanced-usage/containers/docker/index.md

You can build this customised analysis environment image and give it some name, for example `johndoe/myenv`:

```{ .console .copy-to-clipboard }
$ docker build -f environment/myenv/Dockerfile -t johndoe/myenv:1.0 .
```

and push the created image to the DockerHub image registry:

```{ .console .copy-to-clipboard }
$ docker push johndoe/myenv:1.0
```

## Providing necessary shell

The Docker images for executing user jobs in the REANA ecosystem need to
contain `bash` shell in the image.

The `bash` shell is used in operational procedures to pass along
encoded/decoded job commands and parameters between REANA workflow
orchestration components, the job execution components and the compute backend
itself, so that the job execution behaviour would be consistent across
Kubernetes, HTCondor, Slurm backends for both Docker and Singularity execution
wrappers.

Therefore, please make sure that your Docker images contain the `bash` shell
executable, even if it may not be the default shell.

> For example, if you would ilke to use the tiny Alpine image, which uses `ash`

Reviewer: ```suggestion
For example, if you would like to use the tiny Alpine image, which uses `ash`
```

> shell by default, you can add `bash` in your `Dockerfile` as follows:

Reviewer: ```suggestion
shell by default, you can add a command in your `Dockerfile` to install `bash` as follows:
```
WDYT?

```{ .Dockerfile .copy-to-clipboard }
FROM alpine:3.17
RUN apk add bash
```

> The `bash` shell is relatively wide spread, so your base images might already

Reviewer: ```suggestion
The `bash` shell is relatively widespread, so your base images might already
```

contain it. If the requirement on the `bash` presence causes troubles, and you

> cannot add it to your images, then please get in touch.

Reviewer: What about _"Please get in touch with us if requiring `bash` to be present in your Docker images causes you any trouble."_ ?

Author: I have reformulated the whole final paragraph to make it more easily readable as a "conclusion thoughts".
